src/go/configgenerator/clustergen/helpers/backend.go

```diff
@@ -0,0 +1,104 @@
+package helpers
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/GoogleCloudPlatform/esp-v2/src/go/util"
+	clusterpb "github.com/envoyproxy/go-control-plane/envoy/config/cluster/v3"
+	corepb "github.com/envoyproxy/go-control-plane/envoy/config/core/v3"
+	"google.golang.org/protobuf/types/known/durationpb"
+	wrappers "google.golang.org/protobuf/types/known/wrapperspb"
+)
+
+// BackendCluster is an Envoy cluster to communicate with a backend.
+//
+// This should NOT be used directly.
+// Use it via an abstraction like RemoteBackendCluster or LocalBackendCluster.
+type BackendCluster struct {
+	ClusterName string
+	Hostname    string
+	Port        uint32
+	UseTLS      bool
+	Protocol    util.BackendProtocol
+
+	ClusterConnectTimeout  time.Duration
+	MaxRequestsThreshold   int
+	BackendDnsLookupFamily string
+
+	DNS *ClusterDNSConfiger
+	TLS *ClusterTLSConfiger
+}
+
+// GetName implements the ClusterGenerator interface.
```

Comment:
remove? As it doesn't need to implement ClusterGenerator. Its owner will be responsible for that.

Reply:
Sure, done